components/script/dom/bindings/structuredclone.rs

```diff
@@ -67,16 +67,62 @@ unsafe fn read_length(r: *mut JSStructuredCloneReader)
   return length as usize;
 }
 
+struct StructuredCloneWriter {
+    w: *mut JSStructuredCloneWriter,
+}
+
+impl StructuredCloneWriter {
+    unsafe fn write_slice(&self, v: &[u8]) {
+        // writers bytesl takes in a vec! and its length
+        let blob_length = v.len();
+        assert!(JS_WriteBytes(self.w, v.as_ptr() as *const raw::c_void, blob_length));
+    }
+    unsafe fn write_usize(&self, mut u: &usize) {
+        write_length(self.w, u.to_be());
+    }
+}
+
+struct StructuredCloneReader {
+    r: *mut JSStructuredCloneReader,
+}
+
+impl StructuredCloneReader {
+    unsafe fn read_bytes(&self) -> usize {
```

Comment:
`read_bytes` is a fundamental operation that should:
* read a length value N
* create a vector of size N
* read N bytes into that vector
* return the vector